plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHttpMetastoreConfig.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.metastore.thrift;
+
+import com.google.common.base.Joiner;
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.configuration.ConfigSecuritySensitive;
+import jakarta.validation.constraints.NotNull;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class ThriftHttpMetastoreConfig
+{
+    public enum AuthenticationMode
+    {
+        BEARER_TOKEN
+    }
+
+    private String httpBearerToken;
+    private Map<String, String> additionalHeaders = new HashMap<>();
+    private AuthenticationMode authenticationMode;
+
+    public String getHttpBearerToken()
+    {
+        return httpBearerToken;
+    }
+
+    @Config("hive.metastore.http.client.bearer-token")
+    @ConfigSecuritySensitive
+    @ConfigDescription("Http Bearer token to authenticate with a HTTP transport based metastore service")
```

Comment:
nit: No need to repeat "HTTP". 
```suggestion
    @ConfigDescription("Bearer token to authenticate with a HTTP transport based metastore service")
```